File: backend/testTryOn.mjs
import { Client } from '@gradio/client';
import axios from 'axios';
import fs from 'fs';
import path from 'path';
import { execFile } from 'child_process';
import util from 'util';

const execFileAsync = util.promisify(execFile);
const HF_TOKEN = process.env.HF_TOKEN;
const TEST_PERSON = './uploads/test-person.jpg';
const TEST_GARMENT_URL = 'https://placehold.co/300x400/png?text=Blue+Jeans';

async function main() {
  if (!HF_TOKEN) {
    console.error('Missing HF_TOKEN env var. Set HF_TOKEN before running this script.');
    process.exit(1);
  }

  // Download person image if not present
  if (!fs.existsSync('./uploads')) fs.mkdirSync('./uploads');
  if (!fs.existsSync(TEST_PERSON)) {
    console.log('Downloading test person image...');
    const res = await axios.get('https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?w=400', { responseType: 'arraybuffer' });
    fs.writeFileSync(TEST_PERSON, res.data);
    console.log('Downloaded test person image.');
  }

  // Download garment image
  const garmentPath = './uploads/test-garment.jpg';
  if (!fs.existsSync(garmentPath)) {
    console.log('Downloading garment image...');
    const res = await axios.get(TEST_GARMENT_URL, { responseType: 'arraybuffer' });
    fs.writeFileSync(garmentPath, res.data);
    console.log('Downloaded garment image.');
  }

  console.log('\n--- Calling Python Worker ---');
  
  try {
    const { stdout, stderr } = await execFileAsync('python', [
      'ai_tryon.py',
      HF_TOKEN,
      TEST_PERSON,
      garmentPath,
      'blue denim jeans',
      './uploads/test-result.jpg'
    ]);
    console.log('Python Worker stdout:', stdout);
    if(stderr) console.error('Python Worker stderr:', stderr);
    
    const pyResult = JSON.parse(stdout);
    if(pyResult.success){
        console.log('✅ Result saved to ./uploads/test-result.jpg');
    } else {
        console.error('Python worker returned error:', pyResult.error);
    }
  } catch(e) {
    console.error('Python execution failed:', e);
  }
}

main();

File: backend/testTryOn.py
import os
import sys
from gradio_client import Client, file

def test():
    hf_token = os.environ.get("HF_TOKEN")
    if not hf_token:
        raise SystemExit("Missing HF_TOKEN env var. Set HF_TOKEN before running this script.")
    
    person_img = './uploads/test-person.jpg'
    garment_img = './uploads/test-garment.jpg'
    
    print("Connecting to IDM-VTON...")
    client = Client("yisol/IDM-VTON", token=hf_token)
    
    print("Calling API...")
    try:
        result = client.predict(
            dict={"background": file(person_img), "layers": [], "composite": None},
            garm_img=file(garment_img),
            garment_des="blue denim jeans",
            is_checked=True,
            is_checked_crop=False,
            denoise_steps=30,
            seed=42,
            api_name="/tryon"
        )
        print("Success:", result)
    except Exception as e:
        print("Error:", str(e))

if __name__ == '__main__':
    test()
